src/repository.js

```diff
@@ -39,30 +40,89 @@ module.exports = class Repository {
     return entities
   }
 
+  /** 
+  *
+  * Find all method
+  * 
+  * @param {type}   object.limit Limit items to list  
+  * @param {type}   object.orderBy Order by query
+  *
+  * @return {type} List of entities
+  */
+  async findAll(options = {
+    limit: 0,
+    orderBy: null
+  }) {
+
+    const entities = this.find({ limit: options.limit, orderBy: options.orderBy })
+    return entities
+  }
+
+  /** 
+  *
+  * Find with search term
+  * 
+  * @param {type}   search Term to find entities
+  *
+  * @return {type} Single entity
+  */
   async findBy(search) {
+    const entities = await this.find({ search });
+    return entities;
+  }
+
+  /** 
+  *
+  * Find entities
+  * 
+  * @param {type}   object.limit Limit items to list  
+  * @param {type}   object.offset Offset records
+  * @param {type}   object.search Where query term
+  * @param {type}   object.orderBy Order by query
+  *
+  * @return {type} List of entities
+  */
+  async find(options = {
+    limit: 0,
+    offset,
+    orderBy: null,
```

Comment:
How order by `ASC | DESC` (must have) and `NULLS FIRST | LAST` (nice to have) is solved?

Reply:
@dalssoft, to sort `ASC | DESC` you can use the `order` parameter, as in the example below. In the future, we can include the `nullLast` parameter in the orderBy options for` NULLS FIRST | LAST`.

`const usuariosOrderBy = await usuarioRepository.find ({orderBy: [{column: 'last_access', order: 'desc', nullLast: true}]})`

Reply:
great. just make sure it is documented